src/hotspot/share/jeandle/jeandleCompiledCode.cpp

```diff
@@ -437,11 +433,138 @@ static VMReg resolve_vmreg(const StackMapParser::LocationAccessor& location, Sta
   return nullptr;
 }
 
-OopMap* JeandleCompiledCode::build_oop_map(StackMapParser::record_iterator& record) {
+void JeandleCompiledCode::fill_one_scope_value(const StackMapParser& stackmaps, const DeoptValueEncode& encode, const StackMapParser::LocationAccessor& location,
+  GrowableArray<ScopeValue*>* array, int& index) {
+  assert(array != nullptr, "sanity");
+  bool is_constant = StackMapUtil::is_constant(location);
+  switch (encode._basic_type) {
+  case T_INT: {
+    if (is_constant) {
+      array->at_put_grow(index++, new ConstantIntValue(StackMapUtil::getConstantUint(stackmaps, location)));
+    } else {
+      array->at_put_grow(index++,
+        StackMapUtil::is_stack(location)
+        ? new LocationValue(Location::new_stk_loc(Location::Type::normal, StackMapUtil::stack_offset(location)))
+        : new LocationValue(Location::new_reg_loc(Location::Type::normal, resolve_vmreg(location, location.getKind())))
+      );
+    }
+    break;
+  }
+  case T_LONG: {
+    // 2 stack slots for long type in LP64
+    array->at_put_grow(index++, new ConstantIntValue((jint)0));
+    if (is_constant) {
+      array->at_put_grow(index++, new ConstantLongValue(StackMapUtil::getConstantUlong(stackmaps, location)));
+    } else {
+      array->at_put_grow(index++,
+        StackMapUtil::is_stack(location)
+        ? new LocationValue(Location::new_stk_loc(Location::Type::lng, StackMapUtil::stack_offset(location)))
+        : new LocationValue(Location::new_reg_loc(Location::Type::lng, resolve_vmreg(location, location.getKind())))
+      );
+    }
+    break;
+  }
+  case T_FLOAT: {
+    if (is_constant) {
+      array->at_put_grow(index++, new ConstantIntValue(jint_cast(StackMapUtil::getConstantFloat(stackmaps, location))));
+    } else {
+      array->at_put_grow(index++,
+        StackMapUtil::is_stack(location)
+        ? new LocationValue(Location::new_stk_loc(Location::Type::normal, StackMapUtil::stack_offset(location)))
+        : new LocationValue(Location::new_reg_loc(Location::Type::normal, resolve_vmreg(location, location.getKind())))
+      );
+    }
+    break;
+  }
+  case T_DOUBLE: {
+    // 2 stack slots for double type in LP64
```

Comment:
```suggestion
    // 2 stack slots for double type
```
Maybe not just LP64. It's defined by JVM spec for all platforms.